processor/batchprocessor/batch_processor.go

```diff
@@ -274,19 +274,19 @@ func newBatchMetrics(nextConsumer consumer.Metrics) *batchMetrics {
 
 func (bm *batchMetrics) export(ctx context.Context, sendBatchMaxSize int) error {
 	var req pdata.Metrics
-	if sendBatchMaxSize > 0 && bm.metricCount > sendBatchMaxSize {
+	if sendBatchMaxSize > 0 && bm.dataPointCount > sendBatchMaxSize {
 		req = splitMetrics(sendBatchMaxSize, bm.metricData)
-		bm.metricCount -= sendBatchMaxSize
+		bm.dataPointCount -= sendBatchMaxSize
```

Comment:
Is this correct?  The README seems to indicate that a split batch may contain less than the maximum number of data points if the next metric would put the batch over the cap.  I'd expect `bm.dataPointCount` to be reduced here by the number of data points in `req`.

Reply:
Great catch, that is something that could have become very hard to debug issue.
I've created a fix together with a test to verify that is working correctly.

Reply:
Is it not the case that after `splitMetrics` the returned object always has `sendBatchMaxSize`?

Reply:
In this case no.
It boils down to the fact that some metrics have more than one data point and `splitMetrics` not splitting the last metric to fill the batch fully.

Reply:
Can you add a test for when the num points for a metric is greater than `sendBatchMaxSize`. I think we should split the metric as well.

Reply:
Added the logic that splits the metric by data points and the test that you described.
Additionally I changed the Readme to reflect the logic change.
@bogdandrutu PTAL